core_data_modules/traced_data/io.py

```diff
@@ -314,6 +314,97 @@ def import_coda_2_to_traced_data_iterable(cls, user, data, message_id_key, schem
                         Metadata(user, Metadata.get_call_location(), time.time())
                     )
 
+    @classmethod
+    def import_coda_2_to_traced_data_iterable_multi_coded(cls, user, data, message_id_key, scheme_keys, f=None):
+        """
+        Codes keys in an iterable of TracedData objects by using the codes from a Coda 2 messages JSON file.
+
+        Data which is has not been checked in the Coda file is coded using the provided nr_label
+        (irrespective of whether there was an automatic code there before).
+        Data which was previously coded as TRUE_MISSING, SKIPPED, or NOT_LOGICAL by any means is untouched.
+
+        TODO: Data which has been assigned a code under one scheme but none of the others needs to coded as NC not NR
```

Comment:
This is coded as NC with respect to the duplicate fo the scheme right? With respect to the scheme that it is coded with, it will pick up the code. (My this stuff is tricky to describe in words)

Reply:
This TODO is getting at the problem of what to do when we have multiple distinct schemes for the same dataset (e.g. yes/no and reason), but only assigned a code to one of them. Should a code set under one scheme imply anything about which code should be set in the other schemes? We've seen on projects that sometimes it should, but these importers are not currently considering that in the code.

Reply:
Right - I agree with not consider that in this code yet, it seems context specific